src/pyhf/infer/calculators.py

```diff
@@ -168,3 +174,75 @@ def _false_case():
                 (sqrtqmu_v < self.sqrtqmuA_v), _true_case, _false_case
             )
         return teststat
+
+
+class EmpiricalDistribution(object):
+    def __init__(self, samples):
+        self.samples = samples.ravel()
+
+    def pvalue(self, value):
+        tensorlib, _ = get_backend()
+        return (
+            tensorlib.where(self.samples >= value, 1, 0).sum()
+            / tensorlib.shape(self.samples)[0]
+        )
+
+    def expected_value(self, nsigma):
+        tensorlib, _ = get_backend()
+        import numpy as np
```

Comment:
could be hardcoded for the few nsigma values we need, avoiding extension to tensorlib and importing np

Reply:
hard-coding still won't get rid of dependence on `np.percentile` or making `tensorlib.percentile`. The hard-coding on `tensorlib.normal_cdf` should be possible -- although I should point out that nothing needs `*Distribution.expected_value` right now...

Reply:
Question RE: this here: https://github.com/scikit-hep/pyhf/issues/815#issuecomment-609107946

Reply:
what do we do with this? wait for torch 1.7.0 or wrap/accept this use of `default_backend` / `np`. Alternatively we implement percentile ourselves.

Reply:
@lukasheinrich I would vote that we just wrap this for the time being and then come back and remove it when we can, but I don't think we should hold this up any longer than I already have by asking us to wait for other things.

Reply:
See #817.